src/v/cluster/controller_log_limiter.h

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/commands.h"
+#include "config/configuration.h"
+#include "ssx/metrics.h"
+#include "ssx/sformat.h"
+#include "utils/token_bucket.h"
+#include "vlog.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/sharded.hh>
+#include <seastar/core/sstring.hh>
+
+#include <utility>
+
+namespace cluster {
+
+inline ss::logger controller_rate_limiter_log("controller_rate_limiter_log");
+
+struct limiter_configuration {
+    config::binding<bool> enable;
+    config::binding<size_t> topic_rps;
+    config::binding<size_t> topic_capacity;
+    config::binding<size_t> acls_and_users_rps;
+    config::binding<size_t> acls_and_users_capacity;
+    config::binding<size_t> node_management_rps;
+    config::binding<size_t> node_management_capacity;
+    config::binding<size_t> move_rps;
+    config::binding<size_t> move_capacity;
+    config::binding<size_t> configuration_rps;
+    config::binding<size_t> configuration_capacity;
+    config::binding<size_t> internal_rps;
+};
+
+class group_limiter {
```

Comment:
do we need all the methods of limiter to be in header file ?

Reply:
moved